pkg/providers/instance/instance.go

```diff
@@ -59,25 +60,30 @@ type Provider interface {
 }
 
 type DefaultProvider struct {
-	ecsClient *ecsclient.Client
-	region    string
-
+	ecsClient           *ecsclient.Client
+	region              string
 	imageFamilyResolver imagefamily.Resolver
 	vSwitchProvider     vswitch.Provider
 	ackProvider         ack.Provider
+
+	instanceM sync.Map
```

Comment:
How about use `"github.com/patrickmn/go-cache"`?

If it's out of data, then refresh it with list API